tools/src/citation_fetcher.py

```diff
@@ -0,0 +1,131 @@
+from datetime import datetime
+import requests
+from time import sleep
+
+
+def get_all_paper_ids(filename):
+    all_ids = []
+    with open(filename, "r") as file:
+        for line in file:
+            if line.startswith("| "):
+                # Relevant value is in the fourth row
+                paper_id = line.split("|")[4].strip()
+                if len(paper_id) == 40:
+                    all_ids.append(paper_id)
+    return all_ids
+
+
+def fetch_citation_info_from_ids(paper_ids, api_key):
+    results = {}
+    request_headers = {
+        "x-api-key": api_key,
+    }
+    max_tries = 10
+    wait_after_failure = 5  # seconds
+
+    for paper_id in paper_ids:
+        print(f"Now processing paper {paper_id}...")
+        current_attempt = 0
+        success = False
+
+        while not success and current_attempt < max_tries:
+            try:
+                total_citation_count = get_total_citation_count(paper_id, request_headers)
+                if total_citation_count <= 1000:
```

Comment:
I'd propose to make the 1000 a global variable to avoid error-prone repetition in the code.